end_to_end_tests/end_to_end_test_helpers.py

```diff
@@ -0,0 +1,267 @@
+import importlib
+import os
+import re
+import shutil
+from filecmp import cmpfiles, dircmp
+from pathlib import Path
+import sys
+import tempfile
+from typing import Any, Callable, Dict, Generator, List, Optional, Set, Tuple
+
+from attrs import define
+import pytest
+from click.testing import Result
+from typer.testing import CliRunner
+
+from openapi_python_client.cli import app
+from openapi_python_client.utils import snake_case
+
+
+@define
+class GeneratedClientContext:
+    """A context manager with helpers for tests that run against generated client code.
+    
+    On entering this context, sys.path is changed to include the root directory of the
+    generated code, so its modules can be imported. On exit, the original sys.path is
+    restored, and any modules that were loaded within the context are removed.
+    """
+
+    output_path: Path
+    generator_result: Result
+    base_module: str
+    monkeypatch: pytest.MonkeyPatch
+    old_modules: Optional[Set[str]] = None
+
+    def __enter__(self) -> "GeneratedClientContext":
+        self.monkeypatch.syspath_prepend(self.output_path)
+        self.old_modules = set(sys.modules.keys())
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self.monkeypatch.undo()
+        for module_name in set(sys.modules.keys()) - self.old_modules:
+            del sys.modules[module_name]
```

Comment:
Btw, the proof that this "temporarily change the path & modules" stuff works is simply that running all the tests in `generated_code_live_tests` in a single pytest run passes. Since many of the test classes reuse the same generated modules and class names for different things, if the sandboxing didn't work then the tests would inevitably pollute each other and fail.